salt/_modules/metalk8s_volumes.py

```diff
@@ -260,6 +308,25 @@ def format(self, force=False):
         # We format a "normal" file, not a block device: we need force=True.
         super(SparseLoopDevice, self).format(force=True)
 
+    @property
+    def is_cleaned_up(self):
+        return not (self.is_provisioned or self.exists)
+
+    def clean_up(self):
+        LOOP_CLR_FD = 0x4C01  # From /usr/include/linux/loop.h
```

Comment:
Is this portable across archs?

Reply:
arch as in `endianness`?

I think so, there is no `#define` around the constant in the header.

Reply:
I've just tried on big endian (Debian on MIPS) and it works:

```
root@debian:~# truncate -s 1G foo
root@debian:~# losetup -f foo
[   79.088643] loop: module loaded
root@debian:~# losetup -l
NAME       SIZELIMIT OFFSET AUTOCLEAR RO BACK-FILE DIO LOG-SEC
/dev/loop0         0      0         0  0 /root/foo   0     512
root@debian:~# python2
Python 2.7.16 (default, Apr  6 2019, 01:42:57)
[GCC 8.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.machine()
'mips'
>>> import sys
>>> sys.byteorder
'big'
>>> import os
>>> import fcntl
>>> fd = os.open('/dev/loop0', os.O_RDONLY)
>>> fcntl.ioctl(fd, 0x4C01, 0)
0
>>> os.close(fd)
>>>
root@debian:~# losetup -l
root@debian:~#
```

Reply:
:100: